### DISCUSSION TITLE
Pages Beyond 10?

### DISCUSSION THREAD

**Discussion by merlinlcb** [streamdeck-linux-gui/streamdeck-linux-gui#69] [Q&A]
ive been dabbling with adding additional pages to the streamdeck(ive filled all of mine and need more) i built a webpage so that adding them became easier, see [A.P.E.](https://ape.merlinlcb.com/) i can make the stream deck render the pages by doing this however once the pages are rendered it appears i cannot perform any functions using those pages? is there some kind of hard limit? i couldnt find any explanation

**Comment by Danie10** [comment]
It was partly discussed in this issue at https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/7 (it maybe needed a clearer subject line though describing it). It is just the GUI space that limits it (and how that layout would work).

There is also another possibility (possibly complimenting this unlimited option): To have command line options to call a specific config on startup. So you could fire up one session with a general usage and 10 pages, and have some buttons to set to fire up a different configuration with its 10 pages, and a button to return to the general config. I do this currently with Page 1 of my config, where the buttons there open other pages for specific views, with the top left button as a Home button to return to Page 1. I have one page that goes two levels deep, so it is Home, Page 4, then Page 5, returns to Page 4, returns to Home.

The option to call up alternative configurations by button, would effectively give unlimited pages, but it would still be limited to 10 at a time right now. But it could be used by anyone as quickly as we could start up the service using a specified configuration file.

The true unlimited pages will just need some thought around the GUI management of it. Possibly being able to scroll 10 pages up or down, at a time. In other words 1-10 is visible by default, if there is 11-20 pages, then a click down would show 11-20 on that same row, and another click shows 1-10 again.

**Comment by Danie10** [comment]
@4s3ti I see this was just released in the new Dev version - adds pages fine, but the Switch Page setting is still limited to 10, so needs to be increased. It still stops at 10, so you can't tell a button on page 3 to open page 11.

**Reply by coolapso** [reply]
Thanks for the feedback. @rkmax any clues?

**Reply by rkmax** [reply]
thank you for the feedback @Danie10 good catch, I'll fix it

**Reply by rkmax** [reply]
https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/pull/76
